rbs/MethodTypeToParserNode.cc

```diff
@@ -256,49 +275,51 @@ unique_ptr<parser::Node> MethodTypeToParserNode::methodSignature(const parser::N
-        sigBuilder = parser::MK::Send(typeLoc, move(sigBuilder), core::Names::params(), typeLoc, move(args));
+        sigBuilder = parser::MK::Send(fullTypeLoc, move(sigBuilder), core::Names::params(), fullTypeLoc, move(args));
     }
 
     rbs_node_t *returnValue = functionType->return_type;
     if (returnValue->type == RBS_TYPES_BASES_VOID) {
-        auto loc = locFromRange(typeLoc, returnValue->location->rg);
-        sigBuilder = parser::MK::Send0(typeLoc, move(sigBuilder), core::Names::void_(), loc);
+        auto loc = declaration.typeLocFromRange(returnValue->location->rg);
+        sigBuilder = parser::MK::Send0(fullTypeLoc, move(sigBuilder), core::Names::void_(), loc);
     } else {
-        auto returnType = typeToParserNode.toParserNode(returnValue, typeLoc);
+        auto nameLoc = declaration.typeLocFromRange(returnValue->location->rg);
+        auto returnType = typeToParserNode.toParserNode(returnValue, declaration);
         sigBuilder =
-            parser::MK::Send1(typeLoc, move(sigBuilder), core::Names::returns(), returnType->loc, move(returnType));
+            parser::MK::Send1(fullTypeLoc, move(sigBuilder), core::Names::returns(), nameLoc, move(returnType));
     }
 
     auto sigArgs = parser::NodeVec();
-    auto t = parser::MK::T(typeLoc);
-    auto t_sig = parser::MK::Const(typeLoc, move(t), core::Names::Constants::Sig());
-    auto t_sig_withoutRuntime = parser::MK::Const(typeLoc, move(t_sig), core::Names::Constants::WithoutRuntime());
-    sigArgs.emplace_back(move(t_sig_withoutRuntime));
+    sigArgs.emplace_back(assembleTSigWithoutRuntime(firstLineTypeLoc));
 
     auto final = absl::c_find_if(annotations, [](const Comment &annotation) { return annotation.string == "final"; });
     if (final != annotations.end()) {
         sigArgs.emplace_back(parser::MK::Symbol(final->typeLoc, core::Names::final_()));
     }
 
-    auto sig =
-        parser::MK::Send(typeLoc, parser::MK::SorbetPrivateStatic(typeLoc), core::Names::sig(), typeLoc, move(sigArgs));
+    auto sig = parser::MK::Send(fullTypeLoc, parser::MK::SorbetPrivateStatic(fullTypeLoc), core::Names::sig(),
+                                firstLineTypeLoc, move(sigArgs));
 
     return make_unique<parser::Block>(commentLoc, move(sig), nullptr, move(sigBuilder));
 }
 
 unique_ptr<parser::Node> MethodTypeToParserNode::attrSignature(const parser::Send *send, const rbs_node_t *type,
-                                                               const core::LocOffsets typeLoc,
-                                                               const core::LocOffsets commentLoc,
+                                                               const RBSDeclaration &declaration,
                                                                const vector<Comment> &annotations) {
     auto typeParams = vector<pair<core::LocOffsets, core::NameRef>>();
+    // Attribute signatures doesn't support multiple lines
+    // so first line type loc is the same as the full type loc
```

Comment:
fwiw, I could see people wanting long attributes if they have something like

```
T::Hash[
  SomeLongName,
  T.any(Three, Long, Names)
]
```

Reply:
There should probably at least be an error here if someone attempts to use a multiline comment but we're not considering it.